File: 49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res = []
        counter = dict()
        for s in strs:
            if len(s) == 0:
                if "empty" in counter.keys():
                    counter["empty"].append("")
                else:
                    counter["empty"] = [""]
            else:
                key = 0
                for c in s:
                    key += hash(c)
                if key in counter.keys():
                    counter[key].append(s)
                else:
                    counter[key] = [s]
        for key, val in counter.items():
            res.append(val)
        return res
            

File: 49-group-anagrams/49-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> counter = new HashMap<>();
        for (String s : strs) {
            char[] wordArr = s.toCharArray();
            Arrays.sort(wordArr);
            String key = new String(wordArr);
            if (!counter.containsKey(key)) {
                counter.put(key, new ArrayList<String>());
            }
            counter.get(key).add(s);
        }
        List<List<String>> ans = new ArrayList<>();
        for (String key : counter.keySet()) {
            ans.add(counter.get(key));
        }
        return ans;
    }
}